src/command_modules/azure-cli-backup/azure/cli/command_modules/backup/custom.py

```diff
@@ -127,6 +133,16 @@ def list_containers(client, resource_group_name, vault_name, container_type="Azu
     return _get_containers(client, container_type, status, resource_group_name, vault_name)
 
 
+def check_protection_enabled_for_vm(cmd, vm_id):
+    vaults = list_vaults(vaults_cf(cmd.cli_ctx))
+    for vault in vaults:
+        vault_rg = _get_resource_group_from_id(vault.id)
+        items = list_items(cmd, backup_protected_items_cf(cmd.cli_ctx), vault_rg, vault.name)
+        if any(vm_id.lower() == item.properties.virtual_machine_id.lower() for item in items):
+            return vault.id
```

Comment:
Yes, this should be made generic. Today it can support only VM, but in the future, when other resource types are supported, it can easily be extended to handle them without creating a bunch of extra commands.

Reply:
Per our meeting, this is fine as-is. We'll strategize the final command structure later.